pennylane/transforms/commutation_dag.py

```diff
@@ -0,0 +1,1049 @@
+            return np.all(
+                np.allclose(
+                    np.matmul(
+                        qml.Rot(*operation1.data, wires=target_wires_1).get_matrix(),
+                        operation2.get_matrix(),
+                    ),
+                    np.matmul(
+                        operation2.get_matrix(),
+                        qml.Rot(*operation1.data, wires=target_wires_1).get_matrix(),
+                    ),
+                )
+            )
+
+        if operation2.name == "CRot":
+            if not intersection(target_wires_2, operation1.wires):
+                return bool(commutation_map[operation1.name][position["ctrl"]])
+            return np.all(
+                np.allclose(
+                    np.matmul(
+                        qml.Rot(*operation2.data, wires=target_wires_2).get_matrix(),
+                        operation1.get_matrix(),
+                    ),
+                    np.matmul(
+                        operation1.get_matrix(),
+                        qml.Rot(*operation2.data, wires=target_wires_2).get_matrix(),
+                    ),
+                )
+            )
+
+        return np.all(
+            np.allclose(
+                np.matmul(
+                    operation1.get_matrix(),
+                    operation2.get_matrix(),
+                ),
+                np.matmul(
+                    operation2.get_matrix(),
+                    operation1.get_matrix(),
+                ),
+            )
+        )
+    return None
+
+
+not_supported_operations = [
+    "PauliRot",
+    "QubitDensityMatrix",
+    "CVNeuralNetLayers",
+    "ApproxTimeEvolution",
+    "ArbitraryUnitary" "CommutingEvolution",
+    "DisplacementEmbedding",
+    "SqueezingEmbedding",
+]
+non_commuting_operations = [
+    "ArbitraryStatePreparation",
+    "BasisStatePreparation",
+    "MottonenStatePreparation",
+    "QubitCarry",
+    "QubitSum",
+    "SingleExcitation",
+    "SingleExcitationMinus",
+    "SingleExcitationPlus",
+    "DoubleExcitation",
+    "DoubleExcitationPlus",
+    "DoubleExcitationMinus",
+    "BasicEntanglerLayers",
+    "GateFabric",
+    "ParticleConservingU1",
+    "ParticleConservingU2",
+    "RandomLayers",
+    "SimplifiedTwoDesign",
+    "StronglyEntanglingLayers",
+    "AllSinglesDoubles",
+    "FermionicDoubleExcitation",
+    "FermionicSingleExcitation",
+    "Grover",
+    "kUpCCGSD",
+    "Permute",
+    "QFT",
+    "QuantumMonteCarlo",
+    "QuantumPhaseEstimation",
+    "UCCSD",
+    "MPS",
+    "TTN",
+    "AmplitudeEmbedding",
+    "AngleEmbedding",
+    "BasisEmbedding",
+    "IQPEmbedding",
+    "QAOAEmbedding",
+]
+
+
+def is_commuting(operation1, operation2):
+    r"""Check if two operations are commuting. A lookup table is used to check the commutation between the
+    controlled, targetted part of operation 1 with the controlled, targetted part of operation 2. It supports
+    most PennyLane operations that are not CV operations.
```

Comment:
Could be useful to mention in the docstring the qubit operations that are not supported!